docs/frontend-system/utility-apis/04-configuring.md

```diff
@@ -0,0 +1,82 @@
+---
+id: configuring
+title: Configuring Utility APIs
+sidebar_label: Configuring
+# prettier-ignore
+description: Configuring, extending, and overriding utility APIs
+---
+
+Utility APIs are extensions and can therefore optionally be amended with configurability, as well as inputs that other extensions attach themselves to. This section describes how to add those capabilities to your own utility APIs, and how to make use of them as a consumer of such utility APIs.
+
+## Adding configurability
+
+Here we will describe how to amend a utility API with the capability of being configured in app-config. You do this by giving a config schema to your API extension factory function.
+
+Let's make the required additions to [our original work example](./02-creating.md) API.
+
+```tsx title="in @internal/plugin-example"
+import {
+  createApiExtension,
+  createApiFactory,
+  createPlugin,
+  /* highlight-add-next-line */
+  createSchemaFromZod,
+  storageApiRef,
+  StorageApi,
+} from '@backstage/frontend-plugin-api';
+import { WorkApi, workApiRef } from '@internal/plugin-example-react';
+
+/* highlight-add-start */
+interface WorkApiConfig {
+  goSlow: boolean;
+}
+/* highlight-add-end */
+
+const workApi = createApiExtension({
+  api: workApiRef,
+  /* highlight-add-start */
+  configSchema: createSchemaFromZod(z =>
+    z.object({
+      goSlow: z.boolean().default(false),
+    }),
+  ),
+  /* highlight-add-end */
+  /* highlight-remove-next-line */
+  factory: () =>
```

Comment:
So here we'd instead want to highlight that we need to switch to the callback form to be able to access the extension config